packages/react/src/primitives/shared/datastore.ts

```diff
@@ -54,3 +55,19 @@ export const createDataStorePredicate = <Model extends PersistentModel>(
     return predicate;
   };
 };
+
+/**
+ * Creates a DataStore compatible predicate function from an object representation using useCallback
+ * @internal
+ */
+export const useCreateDataStorePredicate = <Model extends PersistentModel>(
+  predicateObject: DataStorePredicateObject
+): ProducerModelPredicate<Model> => {
+  const [predicateObjectState] = React.useState(predicateObject);
```

Comment:
Using `useState` will actually not work here. It will ignore any changes to `predicateObject`.